src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -55,7 +56,8 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                     cachedAccessToken = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
                 }
 
-                if (cachedAccessToken != null)
+                isCachedAccessTokenValid = cachedAccessToken != null && IsCachedAssertionTheSame(cachedAccessToken.Secret);

```

Comment:
Thinking if this filtering could be done in FindAT/FindRT methods in the TokenCache class, but it's probably more efficient to do it here.

Reply:
After discussion - the filtering should probably be done here. Also a method like IsInitiateLROBO called would also be helpful.